includes/razorpay-subscriptions.php

```diff
@@ -0,0 +1,246 @@
+            'quantity'        => (int) $product['qty'],
+            'total_count'     => $length,
+            'customer_notify' => 0,
+            'notes'           => array(
+                'woocommerce_order_id'   => $orderId,
+                'woocommerce_product_id' => $product['product_id']
+            ),
+        );
+
+        $signUpFee = WC_Subscriptions_Product::get_sign_up_fee($product['product_id']);
+
+        if ($signUpFee)
+        {
+            $subscriptionData['addons'] = array(
+                array (
+                    'item' => array(
+                        'amount'   => (int) round($signUpFee * 100),
+                        'currency' => get_woocommerce_currency(),
+                        'name'     => $product['name']
+                    )
+                )
+            );
+        }
+
+        return $subscriptionData;
+    }
+
+    protected function getProductPlanId($product)
+    {
+        $productId = $product['product_id'];
+
+        $metadata = get_post_meta($productId);
+
+        // Creating a new plan only if plan isn't created already
+        // Or if the plan has changed
+        if ($this->shouldCreatePlan($metadata, $productId) === true)
+        {
+            $planId = $this->createPlan($product);
+
+            add_post_meta($productId, 'razorpay_wc_plan_id', $planId, true);
+        }
+        else
+        {
+            // Retrieve the plan id if already created
+            $planId = $metadata['razorpay_wc_plan_id'][0];
+        }
+
+        return $planId;
+    }
+
+    /**
+     * We shouldn't create plan only if the following conditions are met
+     * 1. There exists a plan id as product metadata
+     * 2. The corresponding plan has amount a1
+     * 3. The sale price is amount a2
+     * 4. a1 is not equal to a2, therefore the plan has changed
+     *
+     * @param $metadata
+     * @param $productId
+     */
+    protected function shouldCreatePlan($metadata, $productId)
+    {
+        if (empty($metadata['razorpay_wc_plan_id']) === false)
+        {
+            $planId = $metadata['razorpay_wc_plan_id'][0];
+
+            $plan = $this->api->plan->fetch($planId);
+
+            $salePrice = WC_Subscriptions_product::get_meta_data($productId, 'sale_price', 0) * 100;
+
+            if ((empty($salesPrice) === false) and
+                ($salePrice === $plan['item']['amount']))
+            {
+                return false;
+            }
+
+            //
+            // If we are going to create a new plan id,
+            // we're going to have to delete the old one
+            //
+            delete_post_meta($productId, 'razorpay_wc_plan_id');
+        }
+
+        return true;
+    }
+
+    protected function createPlan($product)
+    {
+        $productId = $product['product_id'];
+
+        $planArgs = $this->getPlanArguments($product);
+
+        $plan = $this->api->plan->create($planArgs);
+
+        // Storing the plan id as product metadata, unique set to true
```

Comment:
@captn3m0 add_post_meta(....., true) on line 127. Ensures that the key is unique. 

https://codex.wordpress.org/Function_Reference/add_post_meta

This ensures that the plan id has to be deleted before we add it again.